File: apps/worldclock.py
"""TinyOLED Desktop — Dünya Saati"""
import time
from core.framebuffer import Framebuffer
ZONES=[("Istanbul",3),("London",0),("New York",-5),("Tokyo",9),("Sydney",10)]

class WorldClockApp:
    NAME="worldclock"; LABEL="Dunya"; ICON="world"
    def __init__(self, on_exit):
        self.on_exit=on_exit; self._cursor=0
    def on_up(self): self._cursor=max(0,self._cursor-1)
    def on_down(self): self._cursor=min(len(ZONES)-1,self._cursor+1)
    def on_sel(self): pass
    def on_long(self): self.on_exit()
    def update(self): pass
    def draw(self, fb: Framebuffer):
        fb.icon("world",1,10); fb.text("Dunya Saati",12,10); fb.hline(0,18,128); y=21
        utc=time.time()
        for i,(name,off) in enumerate(ZONES):
            sel=(i==self._cursor); lt=time.gmtime(utc+off*3600)
            ts=f"{lt.tm_hour:02d}:{lt.tm_min:02d}"
            if sel: fb.rect(0,y-1,128,9,fill=True)
            fb.text(f"{name[:8]:8s} {ts}",2,y,on=not sel); y+=9

File: web-simulator/js/apps/world_clock.js
import { BrowserApp } from '../browser_app.js';
const ZONES=[['IST','Europe/Istanbul'],['LON','Europe/London'],['NYC','America/New_York'],['TYO','Asia/Tokyo'],['SYD','Australia/Sydney']];

export class WorldClockApp extends BrowserApp {
  static NAME='world'; static LABEL='Dunya'; static ICON='world';
  constructor(onExit){super(onExit,{title:'Dunya Saati',mode:'WEB'});this.offset=0;}
  onUp(){this.offset=(this.offset-1+ZONES.length)%ZONES.length;}
  onDown(){this.offset=(this.offset+1)%ZONES.length;}
  onSel(){this.offset=0;}
  draw(fb){
    this.drawHeader(fb,'WEB'); const now=new Date();
    const ordered=[...ZONES.slice(this.offset),...ZONES.slice(0,this.offset)].slice(0,4);
    const rows=ordered.map(([n,z])=>[n,new Intl.DateTimeFormat('tr-TR',{timeZone:z,hour:'2-digit',minute:'2-digit',hour12:false}).format(now)]);
    this.drawRows(fb,rows);this.footer(fb,'UP/DN sehir');
  }
}
